src/Common/ThreadStatus.h

```diff
@@ -107,6 +107,7 @@ using ThreadGroupStatusPtr = std::shared_ptr<ThreadGroupStatus>;
  * - https://github.com/ClickHouse/ClickHouse/pull/40078
  */
 extern thread_local constinit ThreadStatus * current_thread;
+extern thread_local constinit ProfileEvents::Counters * subthread_profile_events;
```

Comment:
Probably it's worth moving this pointer can be moved into `ThreadStatus current_thread`  not to introduce one more thread_local variable.

Reply:
Agree, would be simpler.